### PR TITLE
Abstract token gateway

### LINKED ISSUE #17: [Frontend | Token API Gateway Implementation] Abstract Token API Gateway Singleton
Create an abstract class for a token API gateway.

### PR DESCRIPTION
**Pull request by Jacob4649**
Closes #17.

### CODE REVIEW

File: app/src/main/java/com/javahelp/frontend/gateway/RESTAPITokenGateway.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    /**
+     * Makes a request using JavaHelp authorization.
+     * Overwrites existing Authorization headers
+     *
+     * @param request  {@link SimpleHttpRequest} to make
+     * @param callback {@link FutureCallback} to execute
+     * @return {@link Future} of {@link RESTAPIGatewayResponse<T>}
+     */
+    @Override
+    protected Future<RESTAPIGatewayResponse<T>> getResponse(SimpleHttpRequest request, FutureCallback<RESTAPIGatewayResponse<T>> callback) {
```
Comment (Enkhzul001) [inline app/src/main/java/com/javahelp/frontend/gateway/RESTAPITokenGateway.java:39]: What is Future?

**Review by Enkhzul001 — APPROVED** [review]
Looks good.

**Review by Lam207 — APPROVED** [review]
Good work!

File: app/src/main/java/com/javahelp/frontend/gateway/RESTAPITokenGateway.java (hunk 1 above)
Comment (Jacob4649) [reply app/src/main/java/com/javahelp/frontend/gateway/RESTAPITokenGateway.java:39]: Its like a promise that the type parameter will exist in the future

**Review by Enkhzul001 — APPROVED** [review]
looks good.

**Review by BullDF — APPROVED** [review]

**Pull request merged**